src/main/java/com/microsoft/azure/functions/HttpResponseMessage.java

```diff
@@ -20,15 +20,17 @@
     /**
      * Sets the status code on the HttpResponseMessage instance.
      * @param status An HTTP status code representing the outcome of the HTTP request.
+     * @return this response message
      */
-    void setStatus(int status);
+    HttpResponseMessage<T> setStatus(int status);
```

Comment:
Instead of returning HttpResponseMessage on Set methods, can you add a getValue?

Reply:
Maybe what we should do is have a new Builder class instead. And provide this fluent API through there, and keep the class as is. @JonathanGiles wdyt?

Reply:
@pragnagopa , this style is OK. Similar to the chaining functions (LINQ) in C#.

Reply:
It might be OK in C#, but it is actually not OK in Java. Setter methods are supposed to be void return.

Reply:
I naively implemented this way to keep the API as is, and add the functionality, but thinking better about this, it is a bad practice. I will experiment with the Builder class.

Reply:
Introducing a Builder class is welcomed. Just keep one principal, do make **ALL** implementations in [worker repository](https://github.com/Azure/azure-functions-java-worker) rather than in this repository.

Reply:
Yeap.